internal/internal_decision_state_machine.go

```diff
@@ -485,13 +511,34 @@ func (d *activityCommandStateMachine) handleCancelFailedEvent() {
 	d.failStateTransition(eventCancelFailed)
 }
 
+func (d *activityCommandStateMachine) cancel() {
+	switch d.state {
+	case commandStateCreated, commandStateCommandSent:
+		attribs := &commandpb.RequestCancelActivityTaskCommandAttributes{
+			ScheduledEventId: d.scheduleID,
+		}
+		cancelCmd := d.helper.newCancelActivityStateMachine(attribs)
```

Comment:
Do we create `newCancelActivityStateMachine` state machine only when cancellation happens in the same workflow task?  If that is the case then we should not include `commandStateCommandSent` in the case statement.

Reply:
As it stands without that case we would fail to send the cancellation command if the activity command itself has already been sent. I'm definitely not clear on the implications of being in the same workflow task or not so let's discuss that on our call